src/drivers/magnetometer/bmm150/bmm150.cpp

```diff
@@ -762,7 +770,14 @@ BMM150::ioctl(struct file *filp, int cmd, unsigned long arg)
 
 	default:
 		/* give it to the superclass */
+#if defined(__PX4_POSIX)
```

Comment:
You can use `file_t` instead in the ioctl declaration: `ioctl(file_t *filp, int cmd, unsigned long arg)`.
Then you don't need the ifdef.

Reply:
Did so